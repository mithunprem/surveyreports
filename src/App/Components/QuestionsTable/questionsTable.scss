.questions-table {
  .average-rating-header {
    white-space: nowrap;
  }

  .average-rating-value {
    text-align: center;

    &.text-green{
      color: green;
    }
    &.text-red{
      color: red;
    }
    &.text-yellow{
      color: #d89316;
    }
  }
}
