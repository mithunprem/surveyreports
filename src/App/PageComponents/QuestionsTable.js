import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Table } from 'reactstrap';

const QuestionsTable = ({ questions }) => {
  return (
    <Fragment>
      <Table striped>
        <thead>
          <tr>
            <th>Question</th>
            <th style={{whiteSpace: 'nowrap'}}>Average Rating</th>
          </tr>
        </thead>
        <tbody>
        {
          questions.map(( question, index ) => {
            return (
              <tr key={index}>
                <td>{question.description}</td>
                <td
                  style={{textAlign: 'center'}}
                >{question.averageRating.toFixed(2)}</td>
              </tr>
            )
          })
        }
        </tbody>
      </Table>
    </Fragment>
  );
}

QuestionsTable.propTypes = {
  questions: PropTypes.array
}

export default QuestionsTable;
