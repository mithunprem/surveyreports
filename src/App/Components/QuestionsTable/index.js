import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Table } from 'reactstrap';
import colourCodesForRating from '../../Utils/colourCodesForRating';
import './questionsTable.scss';

const QuestionsTable = ({ questions }) => {
  return (
    <Fragment>
      <Table striped className="questions-table">
        <thead>
          <tr>
            <th>Question</th>
            <th className="average-rating-header">Average Rating</th>
          </tr>
        </thead>
        <tbody>
        {
          questions.map(( question, index ) => {
            const averageRating = question.averageRating.toFixed(2);
            const ratingColourCode = colourCodesForRating(averageRating);
            return (
              <tr key={index}>
                <td>{question.description}</td>
                <td className={`average-rating-value text-${ratingColourCode}`}>
                  {averageRating}
                </td>
              </tr>
            )
          })
        }
        </tbody>
      </Table>
    </Fragment>
  );
}

QuestionsTable.propTypes = {
  questions: PropTypes.array
}

export default QuestionsTable;
