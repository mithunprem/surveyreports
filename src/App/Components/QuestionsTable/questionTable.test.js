import React from 'react';
import { create } from 'react-test-renderer';
import QuestionsTable from '../../Components/QuestionsTable';

describe("QuestionsTable", () => {

  const questions = [
    {description: "Test question 1", averageRating: 3.6},
    {description: "Test question 2", averageRating: 4.6}
  ];

  test("should render without throwing an error", () => {
    const questionsTable = create(<QuestionsTable questions={questions}/>).toJSON();
    expect(questionsTable).toMatchSnapshot();
  });

  test("should render the average rating with appropriate color coding", () => {
    const questionsTable = create(<QuestionsTable questions={questions}/>);
    // First question has a rating of 3.6 and hence this td should have
    // a text-yellow class associated with it.
    const className = questionsTable.root.findAllByType('td')[1].props.className;
    expect(className).toBe('average-rating-value text-yellow');
  });

});
