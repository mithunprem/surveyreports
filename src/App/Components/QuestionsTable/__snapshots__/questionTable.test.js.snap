// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionsTable should render without throwing an error 1`] = `
<table
  className="questions-table table table-striped"
>
  <thead>
    <tr>
      <th>
        Question
      </th>
      <th
        className="average-rating-header"
      >
        Average Rating
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Test question 1
      </td>
      <td
        className="average-rating-value text-yellow"
      >
        3.60
      </td>
    </tr>
    <tr>
      <td>
        Test question 2
      </td>
      <td
        className="average-rating-value text-green"
      >
        4.60
      </td>
    </tr>
  </tbody>
</table>
`;
